src/bp/ui-studio/src/web/reducers/flows.ts

```diff
@@ -698,24 +697,24 @@ reducer = reduceReducers(
         }
       },
 
-      [copyFlowNode as any]: (state, { payload }) => {
-        const nodes =
-          payload &&
-          payload.length &&
-          payload.map(nodeId => _.find(state.flowsByName[state.currentFlow].nodes, { id: nodeId })).filter(node => node)
+      [copyFlowNodes as any]: (state, { payload }) => {
+        const nodes = payload
+          ?.map(nodeId => _.find(state.flowsByName[state.currentFlow].nodes, { id: nodeId }))
+          .filter(node => node)
         if (!nodes || !nodes.length) {
           return state
         }
         return {
           ...state,
-          nodesInBuffer: nodes
+          buffer: { ...state.buffer, nodes }
         }
       },
 
       [requestPasteFlowNode]: (state, { payload: { x, y } }) => {
+        console.log('pasto')
         const currentFlow = state.flowsByName[state.currentFlow]
         const siblingNames = currentFlow.nodes.map(({ name }) => name)
-        const newNodes = _.cloneDeep(state.nodesInBuffer).map(node => {
+        const newNodes = _.cloneDeep(state.buffer.nodes).map(node => {
```

Comment:
Since `buffer.nodes` can be 'null' (e.g. an empty buffer, the first time you reach the flow editor), doing `.map` on a null object will throw an error.

You could maybe do something like this instead:

```ts
const newNodes = _.cloneDeep(state.buffer.nodes || []).map(node => {
```

Reply:
I added a check at the begging of the reducer since there is logic in there that requires a non-empty array

Reply:
I added a check at the begging of the reducer since there is logic in there that requires a non-empty array